tests/unit_tests/dft/include/compute_tester.hpp

```diff
@@ -0,0 +1,120 @@
+/***************************************************************************
+*  Copyright (C) Codeplay Software Limited
+*  Licensed under the Apache License, Version 2.0 (the "License");
+*  you may not use this file except in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+*  For your convenience, a copy of the License has been included in this
+*  repository.
+*
+*  Unless required by applicable law or agreed to in writing, software
+*  distributed under the License is distributed on an "AS IS" BASIS,
+*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*  See the License for the specific language governing permissions and
+*  limitations under the License.
+*
+**************************************************************************/
+
+#ifndef ONEMKL_COMPUTE_TESTER_HPP
+#define ONEMKL_COMPUTE_TESTER_HPP
+
+#if __has_include(<sycl/sycl.hpp>)
+#include <sycl/sycl.hpp>
+#else
+#include <CL/sycl.hpp>
+#endif
+#include "oneapi/mkl.hpp"
+#include "test_helper.hpp"
+#include "test_common.hpp"
+#include "reference_dft.hpp"
+
+template <oneapi::mkl::dft::precision precision, oneapi::mkl::dft::domain domain>
+struct DFT_Test {
+    using descriptor_t = oneapi::mkl::dft::descriptor<precision, domain>;
+
+    template <typename ElemT>
+    using usm_allocator_t = sycl::usm_allocator<ElemT, sycl::usm::alloc::shared, 64>;
+
+    using PrecisionType =
+        typename std::conditional_t<precision == oneapi::mkl::dft::precision::SINGLE, float,
+                                    double>;
+
+    using InputType = typename std::conditional_t<domain == oneapi::mkl::dft::domain::REAL,
+                                                  PrecisionType, std::complex<PrecisionType>>;
+    using OutputType = std::complex<PrecisionType>;
```

Comment:
Yes. We test the forward transform followed by the backward transform. Since we keep the input to the forward transform, there is little reason to compute the inverse for reference purposes.

Reply:
Good idea! I've followed your suggestion in [2fbfcb6](https://github.com/oneapi-src/oneMKL/pull/261/commits/2fbfcb6e6a779ea18e661eb3a6785c497bcad3fd).

Reply:
I would prefer something like `FwdType` and `BwdType`, slightly mimicking the config parameters `FWD_DISTANCE` and `BWD_DISTANCE` which always refer to the same domain for both `compute_forward` and `compute_backward`.